catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/ws/JsonRpc.java

```diff
@@ -0,0 +1,176 @@
+
+  public JsonRpc(Map<String, Function> methods) {
+    this.methods = methods;
+  }
+
+  private static Map<String, Object> parseError(Object message) {
+    return error(-32700, "Parse error", message);
+  }
+
+  private static Map<String, Object> error(int code, String message) {
+    return error(code, message, null);
+  }
+
+  private static Map<String, Object> error(int code, String message, Object data) {
+    Map<String, Object> response = new HashMap<>();
+    response.put("code", code);
+    response.put("message", message);
+    if (data != null) {
+      response.put("data", data);
+    }
+    return response;
+  }
+
+  private static Map<String, Object> success(Object id, Object result) {
+    Map<String, Object> response = new HashMap<>();
+    response.put("jsonrpc", VERSION);
+    response.put("id", id);
+    response.put("result", result);
+    return response;
+  }
+
+  private static Map<String, Object> fail(Object id, Object error) {
+    Map<String, Object> response = new HashMap<>();
+    response.put("jsonrpc", VERSION);
+    response.put("id", id);
+    response.put("error", error);
+    return response;
+  }
+
+  private Map<String, Object> invalid() {
+    return invalid(null);
+  }
+
+  private Map<String, Object> invalid(Object data) {
+    return error(-32600, "Invalid Request", data);
+  }
+
+  private Map<String, Object> invalidParams(Object data) {
+    return error(-32602, "Invalid params", data);
+  }
+
+  @Override
+  public void onOpen(Session session) {}
+
+  @Override
+  public void onClose(Session session, int statusCode, String reason) {}
+
+  @Override
+  public void onError(Session session, Throwable ex) {}
+
+  private Object exec(Object message) {
+
+    if (message instanceof List) {
+      return ((List) message).parallelStream().map(this::exec).collect(Collectors.toList());
+    }
+
+    if (!(message instanceof Map)) {
+      return invalid();
+    }
+
+    Map msg = (Map) message;
+
+    if (!msg.containsKey("id")) {
+      return fail(null, invalid("Required key `id` missing"));
+    }
+
+    Object id = msg.get("id");
+
+    if (!(id instanceof String || id instanceof Number || id == null)) {
+      return fail(null, invalid("Key `id` not string or number or null"));
+    }
+
+    if (!msg.containsKey("jsonrpc")) {
+      return fail(id, invalid("Required key `jsonrpc` missing"));
+    }
+
+    if (!VERSION.equals(msg.get("jsonrpc"))) {
+      return fail(id, invalid("Key `jsonrpc` not equal to `2.0`"));
+    }
+
+    if (!msg.containsKey("method")) {
+      return fail(id, invalid("Required key `method` missing"));
+    }
+
+    if (!(msg.get("method") instanceof String)) {
+      return fail(id, invalid("Key `method` not string"));
+    }
+
+    if (!methods.containsKey(msg.get("method"))) {
+      return fail("id", METHOD_NOT_FOUND);
+    }
+
+    Object params = msg.get("params");
+
+    if (params != null && !(params instanceof List || params instanceof Map)) {
```

Comment:
shouldn't need the params != null, since `null instanceof type` is always false

Reply:
The params key is optional. I think I was using the null check to see if it was set.